Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/BocListValidationResultDispatchingValidatorTest.cs

```diff
@@ -240,5 +242,167 @@ public void EvaluateIsValid_WithFailures_ErrorMessageFromResourceManagerAndInval
       Assert.That(validator.ErrorMessage, Is.EqualTo("BocList has validation errors."));
       Assert.That(validator.IsValid, Is.False);
     }
+
+    [Test]
+    public void DispatchValidationFailures_WithVisibleControl_DispatchesValidationFailuresToRepositoryAndMarkAsHandled ()
+    {
+      var validationResultMock = new Mock<IBusinessObjectValidationResult>();
```

Comment:
For me this test is too complex to understand if it does what it should. There are so mocks and verifications that make the test hard to read. I am unsure on how to best proceed. Maybe @MichaelKetting can chime in here as well.

We want to test that the dispatching uses `control.Visible` in three places:
1. When using `IValidationFailureMatcher`
2. When querying validation failures for the row object
3. When querying validation failures for the list object

The only difference between the `true` and the `false` test is the value so it might be a good idea to use a test parameter instead of duplicating the test. To reduce the amount of mocks I would introduce stub classes or use existing stub classes.

```c#
[Test]
public void METHOD ([Values(true, false)] bool markAsHandled)
{
  var mock = new Mock<IBusinessObjectValidationResult>(MockBehavior.Strict);
  mock.Setup(e => e.GetValidationFailures(It.IsAny<IBusinessObject>(), It.IsAny<IBusinessObjectProperty>(), markAsHandled));
  mock.Setup(e => e.GetUnhandledValidationFailures(It.IsAny<IBusinessObject>(), It.IsAny<bool>(), markAsHandled));

  var validationFailureMatcherMock = new Mock<IValidationFailureMatcher>(MockBehavior.Strict);
  validationFailureMatcherMock.Setup(e => e.GetMatchingValidationFailures(It.IsAny<IBusinessObject>(), mock.Object, markAsHandled));

  var columnDefinitionMock = new Mock<BocColumnDefinition>();
  columnDefinitionMock.As<IBocColumnDefinitionWithValidationSupport>().Setup(e => e.GetValidationFailureMatcher()).Returns(validationFailureMatcherMock.Object);

  var bocListMock = new BocListMock();
  bocListMock.ID = "someBocList";
  bocListMock.Visible = false;
  bocListMock.FixedColumns.Add(columnDefinitionMock.Object);
  // todo setup Business Object and row objects (i would use a similar setup to ManualBusinessObject)

  var validator = new BocListValidationResultDispatchingValidator();
  validator.ControlToValidate = "someBocList";

  NamingContainer.Controls.Add(bocListMock);
  NamingContainer.Controls.Add(validator);

  validator.DispatchValidationFailures(mock.Object);
  
  // we probably want to assert the three scenarios here to ensure that there were any calls since we only check that there are no wrong calls
}
```

By extracting logic into stub I think we can reduce the complexity of the test and make it more readable.